crates/ruff_python_semantic/src/model.rs

```diff
@@ -175,20 +175,13 @@ impl<'a> SemanticModel<'a> {
 
     /// Return `true` if the call path is a reference to `typing.${target}`.
     pub fn match_typing_call_path(&self, call_path: &CallPath, target: &str) -> bool {
-        if call_path.as_slice() == ["typing", target] {
+        if matches!(
+            call_path.as_slice(),
+            ["typing" | "_typeshed" | "typing_extensions", target]
+        ) {
             return true;
         }
 
-        if call_path.as_slice() == ["_typeshed", target] {
-            return true;
-        }
-
-        if is_typing_extension(target) {
```

Comment:
I decided to remove this behavior in which we check if the member is part of `typing_extensions`. We could retain it... but that list may vary over time, so hard-coding it makes Ruff less future-proof.

Reply:
Hm what's the effect of this then? Is it a breaking change?

Reply:
It's not breaking, it's _more_ lax (the net effect is that we consider all imports from `typing_extensions` to be treated equivalently to `typing`, rather than the allow-list that we encoded of specific members). But let's back it out.